stdlib/Statistics/src/Statistics.jl

```diff
@@ -294,22 +350,111 @@ over dimensions, and `m` may contain means for each dimension of `itr`.
     Use the [`skipmissing`](@ref) function to omit `missing` entries and compute the
     variance of non-missing values.
 """
-varm(A::AbstractArray, m::AbstractArray; corrected::Bool=true, dims=:) = _varm(A, m, corrected, dims)
+varm(A::AbstractArray, m; corrected::Bool=true, dims=:,
+     weights::Union{AbstractWeights, Nothing}=nothing) =
+    _varm(A, m, corrected, dims, weights)
+
+varm(iterable, m; corrected::Bool=true) =
+    _var(iterable, corrected, m)
 
-_varm(A::AbstractArray{T}, m, corrected::Bool, region) where {T} =
-    varm!(Base.reducedim_init(t -> abs2(t)/2, +, A, region), A, m; corrected=corrected)
+_varm(A::AbstractArray, m, corrected::Bool, dims, w::Nothing) =
+    varm!(Base.reducedim_init(t -> abs2(t)/2, +, A, dims), A, m, w, corrected=corrected)
 
-varm(A::AbstractArray, m; corrected::Bool=true) = _varm(A, m, corrected, :)
+_varm(A::AbstractArray, m, corrected::Bool, dims, w::AbstractWeights{T}) where {T<:Real} =
+    varm!(Base.reducedim_init(t -> (abs2(t)*zero(T))/2, +, A, dims), A, m, w,
+          corrected=corrected)
 
-function _varm(A::AbstractArray{T}, m, corrected::Bool, ::Colon) where T
+function _varm(A::AbstractArray{T}, m, corrected::Bool, dims::Colon, w::Nothing) where T
     n = length(A)
     n == 0 && return oftype((abs2(zero(T)) + abs2(zero(T)))/2, NaN)
     return centralize_sumabs2(A, m) / (n - Int(corrected))
 end
 
+function _varm(A::AbstractArray{T}, m, corrected::Bool, dims::Colon,
+               w::AbstractWeights) where T
+    s = (zero(T) - zero(m))^2 * zero(eltype(w))
+    @inbounds @simd for i in eachindex(A, w)
+        z = A[i] - m
+        s += (z * z) * w[i]
+    end
+
+    varcorrection(w, corrected) * s
+end
+
+"""
+    varcorrection(n::Integer, corrected=false)
+
+Compute a bias correction factor for calculating `var`, `std` and `cov` with
+`n` observations. Returns ``\\frac{1}{n - 1}`` when `corrected=true`
+(i.e. [Bessel's correction](https://en.wikipedia.org/wiki/Bessel's_correction)),
+otherwise returns ``\\frac{1}{n}`` (i.e. no correction).
+"""
+@inline varcorrection(n::Integer, corrected::Bool=false) = 1 / (n - Int(corrected))
+
+"""
+    varcorrection(w::Weights, corrected=false)
+
+Returns ``\\frac{1}{\\sum w}`` when `corrected=false` and throws an `ArgumentError`
+if `corrected=true`.
+"""
+@inline function varcorrection(w::Weights, corrected::Bool=false)
+    corrected && throw(ArgumentError("Weights type does not support bias correction: " *
+                                     "use FrequencyWeights, AnalyticWeights or ProbabilityWeights if applicable."))
+    1 / w.sum
+end
+
+"""
+    varcorrection(w::AnalyticWeights, corrected=false)
+
+* `corrected=true`: ``\\frac{1}{\\sum w - \\sum {w^2} / \\sum w}``
+* `corrected=false`: ``\\frac{1}{\\sum w}``
```

Comment:
Might be good to include links here as well (e.g., https://en.wikipedia.org/wiki/Weighted_arithmetic_mean)